client/src/components/Bills/BillForm.js

```diff
@@ -94,10 +95,11 @@ const BillForm = (props) => {
           placeholder="0"
           value={bill.tax_total}
           onChange={handleChange}
-          name="tax_amount"
+          name="tax_total"
         />
         <br />
-        Tax Rate: {Math.round(bill.tax_amount / bill.subtotal * 1000000)/10000}%
+        Tax Rate:{" "}
+        {Math.round((bill.tax_amount / bill.subtotal) * 1000000) / 10000}%
```

Comment:
❗ This still uses `bill.tax_amount` ? and I think you've changed it to `bill.tax_total`

Also, the way this is written
```
e.g.
5/30 = 0.1666666667
x 1,000,000 = 166,666.6666666667
Round() = 166,667
/ 10,000 = 16.6667%

```

Is that... I don't know if that's going to be useful?

Tax Rates come in the smallest denominations of like
> New Mexico - approximately  9.0625%

Where 0.0625 is 1/16 of a Percent.  Granted, there's no way of knowing if the tax % is in 1/16th or some arbitrary %.

My Suggestion would be to make a new file in `src/helpers/` called like `caclulator.js` and put in it:

```
// Rounding Introduces all sorts of Mathematical Errors in Javascript
// See this Blog for More: https://www.jacklmoore.com/notes/rounding-in-javascript/
// Warning - this is slow and should not be used in high-performance programs
const roundTo = (value, decimals) => {
  return Number(Math.round(value+'e'+decimals)+'e-'+decimals);
}

// quickly calculates the tax rate (as a percent) to a default of 4 decimals places
const getTaxRate = (tax, bill, decimals = 4) => {
  return roundTo( (tax/bill) * 100, decimals);
}

export { roundTo, getTaxRate } 

```


then

```

import { getTaxRate } from "../../helpers/calculator"

...

        Tax Rate:{" "}
        {getTaxRate(bill.tax_amount , bill.subtotal)}%

```

Reply:
This way the math is only ever in ONE place, and the function name describes what it does instead of whatever 1000000 / 10000 was supposed to do? (I was confused, but the method name `getTaxRate()` is clear.)

Reply:
-tax_amount should be tax_total. Made those changes. Will have to double-check them after merging.
-Going to check out the resource you sent. Might want to talk to you about how to implement those changes.

Reply:
I'm going to work on the helper components later this week.